### DISCUSSION TITLE
Wrap <tr> in component inside pTemplate="header" in p-table

### DISCUSSION THREAD

**Discussion by jasiek-net** [primefaces/community#914] [PrimeNG] [answered]
Hi!
How can I wrap multiple `<tr>` in component and reuse it in pTemplate="header"? For now I have normal header:
```
<ng-template pTemplate="header">
  <tr>
    <th>col1</th>
    <th>col2</th>
  </tr>
  <tr>
    <th>com1</th>
    <th>com2</th>
  </tr>
</ng-template>
```
And I would like to wrap <tr> part into component to achieve smth like this:
```
<ng-template pTemplate="header">
  <header-component></header-component>
</ng-template>
```
It is possible? When I put <tr> into new component it's wrapped with this component selector and the table is broken.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Yes you can, please see the example https://stackblitz.com/edit/zrdgq2?file=src%2Fapp%2Fdemo%2Ftable-basic-demo.ts,src%2Fapp%2Fdemo%2Ftable-basic-demo.html

The technical limitation here is, that you cannot use `ng-template` as host component you should use `tr` for it.

Your table becomes broken because angular injects custom HTML tags by using component selectors and `thead` should be followed by `<tr>` tags not `<custom-header>`.
